### DISCUSSION TITLE
Setting trame-leaflet widget properties with variables

### DISCUSSION THREAD

**Discussion by banesullivan-kobold** [Kitware/trame#347] [Q&A]
I'm trying to track options on the `trame-leaflet` `LMap` widget as state variables, though it seems these are not dynamically updated by the leaflet component. I'm not sure if this is an issue with `trame-leaflet` or `vue-leaflet` but take the following example where I want to have a button that will re-center the `LMap`. I can't get it to work as expected and wondering if anyone has any tips.

```py
import json
import os

from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout
from trame.widgets import html, leaflet, vuetify

server = get_server()
state, ctrl = server.state, server.controller

state.trame__title = "Map Centering"

def recenter(**kwargs):
    state["center"] = (34.792, -111.769)
    # state.dirty("center")
    ctrl.l_map.center = (34.792, -111.769)


with SinglePageLayout(server) as layout:
    layout.title.set_text(state.trame__title)

    with layout.toolbar:
        layout.toolbar.style = "z-index: 1000;"  # really bump it to be above leaflet
        vuetify.VSpacer()

        with vuetify.VBtn(icon=True, click=recenter):
            vuetify.VIcon("mdi-map-marker-outline")

    with layout.content:
        with vuetify.VContainer(
            fluid=True,
            classes="pa-0 fill-height",
        ):
            with leaflet.LMap(zoom=("zoom", 5), center=("center", (34.792, -111.769))) as l_map:
                ctrl.l_map = l_map
                leaflet.LTileLayer(
                    url=(
                        "basemap_url",
                        "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
                    ),
                )


if __name__ == "__main__":
    server.start()

```

**Comment by jourdain** [comment]
So the correct vue way is to use your state variable (like you did), assuming the vue component (LMap) can properly react to it. (maybe a vue-leaflet bug)
Changing it as an attribute on the HTML element after the layout creation will do nothing unless you flush the layout again which could be a way to force the update on the leaflet side. But that is usually overkill. 

To make sure the client gets the new center value, you can print it in the toolbar by adding an `html.Div("{{ center }}")`.
